Rationale.md

```diff
@@ -0,0 +1,216 @@
+# Design Rationale
+
+WebAssembly was designed incrementally, and desisions were made with a solid
+rationale which may turn out to be incorrect in the future. Sometimes, decisions
+were made without the luxury of data, and the designers would like to revisit
+certain decisions once real end-to-end usecases are met and Science can be
+performed.
+
+You, the reader, may also be interested in understanding why
+[Abstract Syntax Tree (AST)](AstSemantics.md) semantics are the way they are,
+and digging through github's issues and pull requests becomes difficult.
+
+This rationale document tries to list how decisions were made, and where
+tradeoffs were made for the sake of language ergonomics, portability,
+performance, security, and Getting Things Done.
+
+[Limited local nondeterminism](Nondeterminism.md) is detailed in its own
+document, its rationale is also explained here.
+
+
+## Why AST?
+
+Why not a stack-, register- or SSA-based bytecode?
+* Trees allow a smaller binary encoding: [JSZap][], [Slim Binaries][].
+* [Polyfill prototype][] shows simple and efficient translation to asm.js.
+
+  [JSZap]: https://research.microsoft.com/en-us/projects/jszap/
+  [Slim Binaries]: https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.108.1711
+  [Polyfill prototype]: https://github.com/WebAssembly/polyfill-prototype-1
+
+
+## Basic Types Only
+
+WebAssembly only represents [a few types](AstSemantics.md#Types).
+* More complex types can be formed from these basic types. It's up to the source
+  language compiler to express its own types in terms of the basic machine
+  types. This allows WebAssembly to present itself as a virtual ISA, and lets
+  compilers target it as they would any other ISA.
+* These types are efficiently executed by all modern architectures.
+* Smaller types (such as `i8` and `i16`) are no more efficient and are only
+  semantically meaningful for memory accesses.
+* Other types (such as `f16`, `f80`, `i128`) aren't widely supported by existing
+  hardware and can be supported by runtime libraries if developers wish to use
+  them. They can be added to WebAssembly later without compromising MVP.
```

Comment:
I'd take `f80` out of this list, or if you really want to mention it, address it separately, because there would be significant hurdles to adding it to WebAssembly later, if it even could be considered.

x86-style `f80` is not portable to non-x86 machines, and even on x86 machines it requires specialized codegen support because of the stack-based register file, so it's a significant implementation burden. And as a soft-float type, it's less desirable than other soft-float types, in part because it has "unsupported" bit patterns sometimes known by names as pseudo-NaNs.